### ISSUE TITLE
Training Pointpillar model for my own dataset but get 0 mAP

### ISSUE THREAD

**Issue by walliampeace** [open-mmlab/mmdetection3d#1039]
Is it my training config file wrong?

**Comment by ZCMax** [comment]
Can you provide your training log, it contains the training loss and evaluation results

**Comment by walliampeace** [comment]
the training log looks like this which seems all 0.
![image](https://user-images.githubusercontent.com/59054117/141209698-44cbcf19-7070-4b18-b2e1-f5a8d9ae6117.png)
The training dataset is converted to Kitti format and preprocessed by running create_data.py script, like this:
![image](https://user-images.githubusercontent.com/59054117/141209820-0cbc7a0e-488a-4234-bb9d-99406f479b25.png)
the training config looks like this, I only changed some parameters like class numbers in dataset config file and I keep the class name to Car and didn't modify it.
![image](https://user-images.githubusercontent.com/59054117/141210820-c364e0d6-caa8-4502-a42c-03dcb507edee.png)
One thing to mention that my dataset training label don't contain 2D bbox values and all set them to 0, is this affecting training?
![image](https://user-images.githubusercontent.com/59054117/141211256-c2f08bf2-7b73-410d-a700-0463bca47d93.png)
Thank you very much!

**Comment by ZCMax** [comment]
2D bbox won't affect training, did your training loss converges well? Or check your calib file?

**Comment by curiousboy20** [comment]
> 2D bbox won't affect training, did your training loss converges well? Or check your calib file?

Hi, I got 0 mAP results on my custom dataset too. How to check if my calib files is okay?

**Comment by 24spiders** [comment]
I want to bump this issue as a 3rd person who has this problem, and provide additional context in the proper format

**Describe the bug**
mAP is 0. This is because the model is not returning _any_ bounding boxes for this users (as well as my own) custom dataset. I cannot speak for the other user, but I have verified my dataset to be in the correct KITTI format. Validation tools provided with this package demonstrate 3D bboxes in the correct location on the pointcloud. No bboxes are output either when training and validating or when testing.

**Reproduction**
1. What command or script did you run?

```
python tools/test.py configs/pointpillars/pp-custom-tree.py work_dirs/pp-custom-tree/epoch_23.pth --show --show-dir ./out/ --out ./out/test.pkl
```

2. Did you make any modifications on the code or config? Did you understand what you have modified?

Yes. A new config was created. However, this is identical to the template config `hv_pointpillars_secfpn_6x8_160e_kitti-3d-car.py`with the following exceptions:
pointcloud_range was changed to reflect the range of my pointcloud. The range of my pointcloud is 
`point_cloud_range=[0, -20, -20, 85, 20, 20,]` . 
This was changed in all the config files that are referenced by this config file ( `hv_pointpillars_secfpn_6x8_160e_kitti-3d-3class.py` as well as 
`    '../_base_/models/hv_pointpillars_secfpn_kitti.py',
    '../_base_/datasets/kitti-3d-3class.py',
    '../_base_/schedules/cyclic_40e.py', '../_base_/default_runtime.py'`
In the `_base_/model` config, Anchor3DRangeGenerator was changed to these values as well to reflect the new required range.

3. What dataset did you use?
A custom dataset, that was verified to work with the visualization tools
**Environment**

1. Please run `python mmdet3d/utils/collect_env.py` to collect necessary environment infomation and paste it here.

**Error traceback**
Here is what is output by the above `test.py` call

```
[                                                  ] 0/6, elapsed: 0s, ETA:####### Printing bbox_results from voxelnet.py ######

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
######################################

[>>>>>>>>                                          ] 1/6, 0.0 task/s, elapsed: 34s, ETA:   169s####### Printing bbox_results from voxelnet.py ######

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
######################################

[>>>>>>>>>>>>>>>>                                  ] 2/6, 0.1 task/s, elapsed: 35s, ETA:    71s####### Printing bbox_results from voxelnet.py ######

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
######################################

[>>>>>>>>>>>>>>>>>>>>>>>>>                         ] 3/6, 0.1 task/s, elapsed: 37s, ETA:    37s####### Printing bbox_results from voxelnet.py ######

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
######################################

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>                 ] 4/6, 0.1 task/s, elapsed: 38s, ETA:    19s####### Printing bbox_results from voxelnet.py ######

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
######################################

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>         ] 5/6, 0.1 task/s, elapsed: 40s, ETA:     8s####### Printing bbox_results from voxelnet.py ######

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
######################################

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 6/6, 0.1 task/s, elapsed: 41s, ETA:     0s(mmcv)
```
Here is what is output when I modify `mmdet3d/apis/test.py` to print the `result` after line 39

```
Use load_from_local loader
[                                                  ] 0/6, elapsed: 0s, ETA:####### Printing bbox_results from voxelnet.py ######

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
######################################


 vvvvvvvvvvResult of single_gpu_test directly: vvvvvvvvvvvv

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
^^^^^^^^^^^^^^^ 

[>>>>>>>>                                          ] 1/6, 0.3 task/s, elapsed: 4s, ETA:    19s####### Printing bbox_results from voxelnet.py ######

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
######################################


 vvvvvvvvvvResult of single_gpu_test directly: vvvvvvvvvvvv

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
^^^^^^^^^^^^^^^ 

[>>>>>>>>>>>>>>>>                                  ] 2/6, 0.4 task/s, elapsed: 5s, ETA:    11s####### Printing bbox_results from voxelnet.py ######

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
######################################


 vvvvvvvvvvResult of single_gpu_test directly: vvvvvvvvvvvv

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
^^^^^^^^^^^^^^^ 

[>>>>>>>>>>>>>>>>>>>>>>>>>                         ] 3/6, 0.4 task/s, elapsed: 7s, ETA:     7s####### Printing bbox_results from voxelnet.py ######

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
######################################


 vvvvvvvvvvResult of single_gpu_test directly: vvvvvvvvvvvv

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
^^^^^^^^^^^^^^^ 

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>                 ] 4/6, 0.5 task/s, elapsed: 8s, ETA:     4s####### Printing bbox_results from voxelnet.py ######

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
######################################


 vvvvvvvvvvResult of single_gpu_test directly: vvvvvvvvvvvv

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
^^^^^^^^^^^^^^^ 

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>         ] 5/6, 0.5 task/s, elapsed: 10s, ETA:     2s####### Printing bbox_results from voxelnet.py ######

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
######################################


 vvvvvvvvvvResult of single_gpu_test directly: vvvvvvvvvvvv

[{'boxes_3d': LiDARInstance3DBoxes(
    tensor([], size=(0, 7))), 'scores_3d': tensor([]), 'labels_3d': tensor([], dtype=torch.int64)}]
^^^^^^^^^^^^^^^ 

[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 6/6, 0.6 task/s, elapsed: 11s, ETA:     0s
```
As discussed, no boxes are drawn! However, Open3D loads and displays the pointclouds no problem; they look correct placing.
Please advise on further steps. Thanks!!

**Comment by Comverser** [comment]
I have the same problem with non bbox data as @walliampeace mentioned on [this](https://github.com/open-mmlab/mmdetection3d/issues/1039#issuecomment-965850791).

I understood @24spiders means this problem is due to absence of bbox data. Is it right?

If I shouldn't write bbox data as "0 0 0 0", what value will be good?

**Comment by Yyb-XJTU** [comment]
Did you finally solve it?

**Comment by Random-junhao** [comment]
Hello, I encountered the same situation as you. During the training process, it seems that the loss is also calculated while training. But the result is always 0 during verification. Have you solved it?

04/17 19:07:04 - mmengine - INFO - Epoch(train)  [1][4900/5139]  lr: 3.0000e-03  eta: 6:05:36  time: 0.1090  data_time: 0.0023  memory: 1763  grad_norm: 2.3234  loss: 3.5920  loss_cls: 0.8903  loss_bbox: 2.5680  loss_dir: 0.1337
04/17 19:07:09 - mmengine - INFO - Epoch(train)  [1][4950/5139]  lr: 3.0000e-03  eta: 6:05:29  time: 0.1089  data_time: 0.0023  memory: 1763  grad_norm: 2.4469  loss: 3.8129  loss_cls: 0.8912  loss_bbox: 2.7879  loss_dir: 0.1338
04/17 19:07:15 - mmengine - INFO - Exp name: mvxnet_fpn_dv_second_secfpn_8xb2-80e_vod-3d-3class_20240417_185757
04/17 19:07:15 - mmengine - INFO - Epoch(train)  [1][5000/5139]  lr: 3.0000e-03  eta: 6:05:23  time: 0.1089  data_time: 0.0023  memory: 1763  grad_norm: 2.3599  loss: 3.6229  loss_cls: 0.8827  loss_bbox: 2.6032  loss_dir: 0.1370
04/17 19:07:20 - mmengine - INFO - Epoch(train)  [1][5050/5139]  lr: 3.0000e-03  eta: 6:05:17  time: 0.1090  data_time: 0.0023  memory: 1763  grad_norm: 2.3603  loss: 3.5812  loss_cls: 0.8593  loss_bbox: 2.5850  loss_dir: 0.1369
04/17 19:07:25 - mmengine - INFO - Epoch(train)  [1][5100/5139]  lr: 3.0000e-03  eta: 6:05:12  time: 0.1094  data_time: 0.0023  memory: 1763  grad_norm: 2.2567  loss: 3.6690  loss_cls: 0.8690  loss_bbox: 2.6639  loss_dir: 0.1361
----------- AP11 Results ------------

Pedestrian AP11@0.50, 0.50, 0.50:
bbox AP11:0.0000, 0.0000, 0.0000
bev  AP11:0.0000, 0.0000, 0.0000
3d   AP11:0.0000, 0.0000, 0.0000
Pedestrian AP11@0.50, 0.25, 0.25:
bbox AP11:0.0000, 0.0000, 0.0000
bev  AP11:0.0000, 0.0000, 0.0000
3d   AP11:0.0000, 0.0000, 0.0000
Cyclist AP11@0.50, 0.50, 0.50:
bbox AP11:0.0000, 0.0000, 0.0000
bev  AP11:0.0000, 0.0000, 0.0000
3d   AP11:0.0000, 0.0000, 0.0000
Cyclist AP11@0.50, 0.25, 0.25:
bbox AP11:0.0000, 0.0000, 0.0000
bev  AP11:0.0000, 0.0000, 0.0000
3d   AP11:0.0000, 0.0000, 0.0000
Car AP11@0.70, 0.70, 0.70:
bbox AP11:0.0000, 0.0000, 0.0000
bev  AP11:0.0000, 0.0000, 0.0000
3d   AP11:0.0000, 0.0000, 0.0000
Car AP11@0.70, 0.50, 0.50:
bbox AP11:0.0000, 0.0000, 0.0000
bev  AP11:0.0000, 0.0000, 0.0000
3d   AP11:0.0000, 0.0000, 0.0000

Overall AP11@easy, moderate, hard:
bbox AP11:0.0000, 0.0000, 0.0000
bev  AP11:0.0000, 0.0000, 0.0000
3d   AP11:0.0000, 0.0000, 0.0000

----------- AP40 Results ------------

Pedestrian AP40@0.50, 0.50, 0.50:
bbox AP40:0.0000, 0.0000, 0.0000
bev  AP40:0.0000, 0.0000, 0.0000
3d   AP40:0.0000, 0.0000, 0.0000
Pedestrian AP40@0.50, 0.25, 0.25:
bbox AP40:0.0000, 0.0000, 0.0000
bev  AP40:0.0000, 0.0000, 0.0000
3d   AP40:0.0000, 0.0000, 0.0000
Cyclist AP40@0.50, 0.50, 0.50:
bbox AP40:0.0000, 0.0000, 0.0000
bev  AP40:0.0000, 0.0000, 0.0000
3d   AP40:0.0000, 0.0000, 0.0000
Cyclist AP40@0.50, 0.25, 0.25:
bbox AP40:0.0000, 0.0000, 0.0000
bev  AP40:0.0000, 0.0000, 0.0000
3d   AP40:0.0000, 0.0000, 0.0000
Car AP40@0.70, 0.70, 0.70:
bbox AP40:0.0000, 0.0000, 0.0000
bev  AP40:0.0000, 0.0000, 0.0000
3d   AP40:0.0000, 0.0000, 0.0000
Car AP40@0.70, 0.50, 0.50:
bbox AP40:0.0000, 0.0000, 0.0000
bev  AP40:0.0000, 0.0000, 0.0000
3d   AP40:0.0000, 0.0000, 0.0000

Overall AP40@easy, moderate, hard:
bbox AP40:0.0000, 0.0000, 0.0000
bev  AP40:0.0000, 0.0000, 0.0000
3d   AP40:0.0000, 0.0000, 0.0000

**Comment by Vish19-code** [comment]
i have same problem i am using kitti dataset in mvxnet but while training all Ap is zero

**Comment by Vish19-code** [comment]
hello you solve it @Tai-Wang @Joaovsky @holtvogt

**Comment by holtvogt** [comment]
Hi @Vish19-code ! Honestly, I can't really remember 100 % how I mitigated this issue. Have you already tried to visualize your dataset using the tools from MMDetection?

**Comment by Vish19-code** [comment]
Hello,

Yes already visualised and it is correct but while training there will be
problem

Can you please guide me what should I do?
